lib/OrdinaryDiffEqLinear/src/algorithms.jl

```diff
@@ -1,34 +1,76 @@
 # Linear Methods
 
-for Alg in [
-    :MagnusMidpoint,
-    :MagnusLeapfrog,
-    :LieEuler,
-    :MagnusGauss4,
-    :MagnusNC6,
-    :MagnusGL6,
-    :MagnusGL8,
-    :MagnusNC8,
-    :MagnusGL4,
-    :RKMK2,
-    :RKMK4,
-    :LieRK4,
-    :CG2,
-    :CG3,
-    :CG4a
-]
-    @eval struct $Alg <: OrdinaryDiffEqLinearExponentialAlgorithm
+for (Alg, Description, Ref) in [
+    (:MagnusMidpoint, "Second order Magnus Midpoint method.", "ref TBD"),
+    (:MagnusLeapfrog, "Second order Magnus Leapfrog method.", "ref TBD"),
+    (:LieEuler, "description", "ref TBD"),
+    (:MagnusGauss4, "Fourth order Magnus method approximated using a two stage Gauss quadrature.", "ref TBD"),
+    (:MagnusNC6, "Sixth order Magnus method approximated using Newton-Cotes quadrature.", "ref TBD"),
+    (:MagnusGL6, "Sixth order Magnus method approximated using Gauss-Legendre quadrature.", "ref TBD"),
+    (:MagnusGL8, "Eighth order Magnus method approximated using Newton-Cotes quadrature.", "ref TBD"),
+    (:MagnusNC8, "Eighth order Magnus method approximated using Gauss-Legendre quadrature.", "ref TBD"),
+    (:MagnusGL4, "Fourth order Magnus method approximated using Gauss-Legendre quadrature.", "ref TBD"),
+    (:RKMK2, "Second order Runge–Kutta–Munthe-Kaas method.", "ref TBD"),
+    (:RKMK4, "Fourth order Runge–Kutta–Munthe-Kaas method.", "ref TBD"),
+    (:LieRK4, "Fourth order Lie Runge-Kutta method.", "ref TBD"),
+    (:CG2, "Second order Crouch–Grossman method.", "ref TBD"),
+    (:CG3, "Third order Crouch-Grossman method.", "ref TBD"),
+    (:CG4a, " Fourth order Crouch-Grossman method.", "ref TBD")]
+    @eval begin
+        @doc generic_solver_docstring($Description,
+            $(string(Alg)),
+            "Semilinear ODE solver",
+            $Ref,
+            """
+            - `krylov`: TBD
+            - `m`: TBD
+            - `iop`: TBD
+            """,
+            """
+            krylov = false,
+            m = 30,
+            iop = 0,
+            """)
+        struct $Alg <: OrdinaryDiffEqLinearExponentialAlgorithm
         krylov::Bool
         m::Int
         iop::Int
+        end
     end
     @eval $Alg(; krylov = false, m = 30, iop = 0) = $Alg(krylov, m, iop)
 end
 
+@doc generic_solver_docstring("Fourth Order Adaptive Magnus method.",
+    "MagnusAdapt4",
+    "Semilinear ODE solver",
+    "ref TBD", "", "")
 struct MagnusAdapt4 <: OrdinaryDiffEqAdaptiveAlgorithm end
 
+@doc generic_solver_docstring("First order method using Cayley transformations.",
+    "CayleyEuler",
+    "Semilinear ODE solver",
+    "ref TBD", "", "")
```

Comment:
@ChrisRackauckas ref

Reply:
I believe this is also in Hairer Geometric.

Reply:
It is this one:
https://www.cambridge.org/core/journals/acta-numerica/article/abs/liegroup-methods/856125FF1EAF7762DEF6E37EEBA9CA5F